### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by HollowPrincess**
Fix `raise_exception` flag in `run_notebook`

Снимок экрана от 2023-08-25 17-33-07

Update `pylint` to 3.0.2 version

### CODE REVIEW

File: nbtools/pylint_notebook.py [hunk 1]
```diff
@@ -101,7 +102,10 @@ def pylint_notebook(path=None, options=(), disable=(), enable=(), printer=print,
     pylint_params : dict
         Additional parameter of linting. Each is converted to a separate valid entry in the `pylintrc` file.
     """
-    from pylint import epylint as lint
+    from io import StringIO
+    from pylint.lint import Run
+    from pylint.reporters.text import TextReporter
```
Comment (SergeyTsimfer) [inline nbtools/pylint_notebook.py:107]: `StringIO` can be at the top of the module

Wrap imports from `pylint` into `try/except ImportError` with readable error message

File: nbtools/pylint_notebook.py [hunk 2]
```diff
@@ -117,25 +121,40 @@ def pylint_notebook(path=None, options=(), disable=(), enable=(), printer=print,
...
 
     # Run pylint on script with pylintrc configuration
-    pylint_cmdline = ' '.join([path_pylintrc, f'--rcfile {path_pylintrc}', *options])
-    pylint_stdout, pylint_stderr = lint.py_run(pylint_cmdline, return_std=True)
+    pylint_cmdline = [path_pylintrc, f'--rcfile={path_pylintrc}', *options]
+
+    # Custom streams for catching messagies
+    pylint_stdout = StringIO()
+
+    base_stderr = sys.stderr
```
Comment (SergeyTsimfer) [inline nbtools/pylint_notebook.py:129]: Can we use `from contextlib import redirect_stdout`?

File: nbtools/run_notebook.py [hunk 3]
```diff
@@ -340,6 +348,11 @@ def run_notebook(path, inputs=None, outputs=None, inputs_pos=1, replace_inputs_p
         # Prepare execution results: execution state, notebook outputs and error info (if exists)
         if failed:
             exec_res = {'failed': failed, 'failed cell number': error_cell_num, 'traceback': traceback_message}
+
+            # Re-raise exception if needed
+            if raise_exception:
+                returned_value.put(exec_res)
```
Comment (SergeyTsimfer) [inline nbtools/run_notebook.py:354]: Please, rename `returned_value` to something `queue` related. `_queue`, `_output_queue`,

File: nbtools/pylint_notebook.py (hunk 2 above)
Comment (HollowPrincess) [reply nbtools/pylint_notebook.py:129]: Thanks, that much better

**Pull request merged**